index.xml

```diff
@@ -3948,6 +3967,101 @@ setlike&lt;<i>type</i>&gt;;</pre>
           </div>
         </div>
 
+        <div id='idl-namespaces' class='section'>
+          <h3>Namespaces</h3>
+
+          <p>
+            A <dfn data-export='' id='dfn-namespace'>namespace</dfn> is a definition (matching <a
+            class='sym' href='#prod-Namespace'>Namespace</a>) that declares a global singleton with
+            associated behaviors.
+          </p>
+
+          <pre class='syntax'>namespace <i>identifier</i> {
+  <i>namespace-members…</i>
+};</pre>
+
+          <p>
+            A namespace is a specification of a set of <dfn data-export='' data-lt='namespace
+            member' id='dfn-namespace-member'>namespace members</dfn> (matching <a class='sym'
+            href='#prod-NamespaceMembers'>NamespaceMembers</a>), which are the <a class='dfnref'
+            href='#dfn-regular-operation'>regular operations</a> that appear between the braces in
+            the namespace declaration. These operations describe the behaviors packaged into the
+            namespace.
+          </p>
+
+          <p>
+            As with interfaces, the IDL for namespaces can be split into multiple parts by using
+            <dfn data-export='' id='dfn-partial-namespace'>partial namespace</dfn> definitions
+            (matching <span class='sym'>"partial"</span> <a class='sym'
+            href='#prod-Namespace'>Namespace</a>). The <a class='dfnref'
+            href='#dfn-identifier'>identifier</a> of a partial namespace definition <span
+            class='rfc2119'>MUST</span> be the same as the identifier of a namespace definition.
+            All of the members that appear on each of the partial namespace definitions are
+            considered to be members of the namespace itself.
+          </p>
+
+          <pre class='syntax'>namespace <em>SomeNamespace</em> {
+  <i>namespace-members…</i>
+};
+
+partial namespace <em>SomeNamespace</em> {
+  <i>namespace-members…</i>
+};</pre>
+
+          <div class='note'>
+            <p>
+              As with partial interface definitions, partial namespace definitions are intended for
+              use as a specification editorial aide, allowing the definition of an interface to be
+              separated over more than one section of the document, and sometimes multiple
+              documents.
+            </p>
+          </div>
+
+          <p>
+            The order that members appear has significance both for <a
+            href='#idl-overloading'>overloading</a> and for property enumeration in the <a
+            href='#es-namespaces'>ECMAScript binding</a>.
+          </p>
+
+          <p>
+            Note that unlike interfaces or dictionaries, namespaces do not create types.
+          </p>
+
+          <p>
+            Only the <a class='xattr' href='#Exposed'>[Exposed]</a> and <a class='xattr'
```

Comment:
This section should be clearer that it's only talking about extended attributes defined by _this_ specification, because we do want console to define an extended attribute that applies to a namespace.